src/crystal/system/unix/user.cr

```diff
@@ -33,17 +26,27 @@ module Crystal::System::User
 
     pwd = uninitialized LibC::Passwd
     pwd_pointer = pointerof(pwd)
-    initial_buf = uninitialized UInt8[1024]
-    buf = initial_buf.to_slice
-
-    ret = LibC.getpwuid_r(id, pwd_pointer, buf, buf.size, pointerof(pwd_pointer))
-    while ret == LibC::ERANGE && buf.size < GETPW_R_SIZE_MAX
-      buf = Bytes.new(buf.size * 2)
-      ret = LibC.getpwuid_r(id, pwd_pointer, buf, buf.size, pointerof(pwd_pointer))
+    User.retry_with_buffer("getpwuid_r", GETPW_R_SIZE_MAX) do |buf|
+      LibC.getpwuid_r(id, pwd_pointer, buf, buf.size, pointerof(pwd_pointer))
     end
 
-    raise RuntimeError.from_errno("getpwuid_r") if ret != 0
-
     from_struct(pwd) if pwd_pointer
   end
+
+  def self.retry_with_buffer(function_name, max_buffer)
```

Comment:
Could you mark this with `# :nodoc:` please?

Reply:
Sure

Reply:
`Crystal::System` namespace is nodoc anyways, so this really isn't necessary.
I was wondering, should we move this method to a more generic location like `retry_wstr_buffer`? It's not specific to user (or group) related methods.

Reply:
So it could be called `Crystal::System.retry_with_buffer` and placed in a `src/crystal/system/unix.cr`.

Reply:
OK, I will move it.